src/vs/workbench/contrib/terminalContrib/suggest/browser/terminalCompletionService.ts

```diff
@@ -166,7 +166,10 @@ export class TerminalCompletionService extends Disposable implements ITerminalCo
 			if (provider.shellTypes && !provider.shellTypes.includes(shellType)) {
 				return undefined;
 			}
-			const completions = await provider.provideCompletions(promptValue, cursorPosition, allowFallbackCompletions, token);
+			const completions = await Promise.race([
+				provider.provideCompletions(promptValue, cursorPosition, allowFallbackCompletions, token),
+				new Promise<undefined>(resolve => setTimeout(() => resolve(undefined), 5000))
```

Comment:
5s is a long time, does the editor suggest widget do something like this across all providers?

We may want to also add the 300ms one on the extension side since we don't need to worry about remote latency on that end?

Reply:
meaning go with 300 ms instead here? or add the 300 ms elsewhere in addition?

Reply:
changing 5000 -> 300. I got 5000 from tasks, that's how long we wait there

Reply:
I'd keep a longer number here, maybe 5s is fine for now. I mean another timeout in the terminal-suggest extension, the one you linked it in the typescript extension. 300ms on the renderer proc is no good as it would fall over on a high latency remote window.